frontend/src/components/smart/ShippingInfoModal.vue

```diff
@@ -0,0 +1,191 @@
+
+export interface Recipient {
+  name: string;
+  email: string;
+  phone: string;
+  country: string;
+  state?: string;
+  address1: string;
+  address2?: string;
+  city: string;
+  zip: string;
+  company: string;
+}
+
+export interface Country {
+  name: string;
+  code: string;
+  states?: State[];
+}
+
+export interface State {
+  name: string;
+  code: string;
+}
+
+interface OrderItem {
+  sync_variant_id: number;
+  product_id: string;
+  quantity: number;
+}
+
+interface StoreMappedActions {
+  purchaseMerchandise(payload: { ids: number[], amounts: number[], totalPrice: BigNumber }): Promise<number>;
+}
+
+interface Data {
+  selectedCountry?: Country;
+  selectedState?: State;
+  countries: Country[];
+  recipient: Recipient;
+  cartEntries: CartEntry[];
+  totalPriceInSkill: number;
+}
+
+export default Vue.extend({
+  data() {
+    return {
+      selectedCountry: undefined,
+      selectedState: undefined,
+      countries: [],
+      recipient: {} as Recipient,
+      cartEntries: [],
+      totalPriceInSkill: 0,
+    } as Data;
+  },
+
+  computed: {
+    emailState() {
+      return this.$data.recipient.email && this.$data.recipient.email.includes('@');
+    },
+    disablePlaceOrderButton() {
+      return !this.$data.recipient.name
+        || !this.$data.recipient.email
+        || !this.$data.recipient.phone
+        || !this.$data.selectedCountry
+        || (this.$data.selectedCountry.states && this.$data.selectedCountry.states.length !== 0 && !this.$data.selectedState)
+        || !this.$data.recipient.address1
+        || !this.$data.recipient.city
+        || !this.$data.recipient.zip;
+    }
+  },
+
+  methods: {
+    ...mapActions(['purchaseMerchandise']) as StoreMappedActions,
+    countryChanged() {
+      this.selectedState = undefined;
+    },
+
+    async fetchCountries() {
+      const response = await api.getMerchandiseCountries();
+      if (response.code !== 200) {
+        return;
+      }
+      this.countries = response.result;
+    },
+
+    async buyItem() {
+      if (!this.selectedCountry) return;
+      this.$root.$emit('merchandise-order-loading', true);
+      this.recipient.country = this.selectedCountry.code;
+      this.recipient.state = this.selectedState?.code;
+      const orderItems = this.cartEntries.map(cartEntry => {
+        return {
+          sync_variant_id: cartEntry.variant.id,
+          quantity: cartEntry.quantity
+        } as OrderItem;
+      });
+      const merchandiseOrder: MerchandiseOrder = {
+        recipient: this.recipient,
+        items: orderItems
+      };
+
+      this.$root.$emit('merchandise-cart-modal', false);
+      try {
+        await this.purchaseMerchandise({
+          ids: merchandiseOrder.items.map(item => item.sync_variant_id),
+          amounts: merchandiseOrder.items.map(item => item.quantity),
+          totalPrice: toBN(this.totalPriceInSkill),
+        });
+        const apiResponse = await api.createMerchandiseOrder(merchandiseOrder);
```

Comment:
Can I just send this request and successfully place an order? Is the payment transaction validated anywhere along the way?

Reply:
It was decided to calculate prices in FE, we need to firstly get the payment and later the request to api is being sent.  Orders come as drafts on the printful side, can be validated there somehow. I only got instructions for that part 👍